### ISSUE TITLE
Decorator output is inconsistent

### ISSUE THREAD

**Issue by vjeux** [eslint/typescript-eslint-parser#250]
In this example, the decorator is not wrapped in a `Decorator` node:

```js
class Person {
  @inject leftLeg: Leg;
}

==>

node.decorators = [ {type: 'Identifier'} ];
```

In this example, it is wrapped in a `TSDecorator` node, instead of `Decorator` node:

```js
@dec enum E {}

==>

node.decorators = [ {type: 'TSDecorator', expression: {type: 'Identifier'}} ];
```

It would be nice if decorators arrays was always wrapped in a `Decorator` node (without `TS` prefix). This way it would work out of the box with babylon decorators.

I have a workaround in prettier, but I'd rather it be fixed inside of this project: https://github.com/prettier/prettier/pull/1509

**Comment by soda0289** [comment]
Thanks for the report.
We can make this change. We should be following the estree spec:
https://github.com/estree/estree/blob/master/experimental/decorators.md

Typescript also allows decorators on method parameters but should still be using the same node type in all cases.

**Comment by vjeux** [comment]
Thanks for being super responsive!

**Comment by JamesHenry** [comment]
Nice to see you over this side @vjeux 😉

Just FYI ```@dec enum E {}``` is semantically invalid TypeScript, there are currently no examples of decorators outside of a `class` context in TS. But regardless, I agree that the AST should be consistent, even when the semantics are incorrect. The raw TypeScript AST definitely is in this case.

I'll submit a PR for this tomorrow.

**Comment by vjeux** [comment]
We're running prettier on all the typescript tests and looking at where it crashes. So expect a lot of crazy edge cases uncovered!

**Comment by vjeux** [comment]
vscode uses a lot of decorators and right now prettier is crashing because of the inconsistency fyi.

**Comment by JamesHenry** [comment]
@vjeux Finally got to this, sorry for the delay https://github.com/eslint/typescript-eslint-parser/pull/293

**Comment by vjeux** [comment]
Nice! Thanks. I added a workaround inside of prettier for now, but i'd love to get rid of it :)
